wait for document loading, then observe body

valdrinkoshi opened the pull request:
Fixes https://github.com/WICG/inert/issues/13 by waiting for the document to load before observing body and adding inert styles.

alice commented on inert.js:
```diff
@@ -388,9 +388,13 @@ class InertManager {
     this._observer = new MutationObserver(this._watchForInert.bind(this));
-    this._observer.observe(document.body, { attributes: true, subtree: true, childList: true });
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', this._observeBody.bind(this));
```
This seems broken - if any inert roots are added in between this constructor being called and the `DOMContentLoaded` event being fired, they won't be correctly set up, right?

Perhaps instead we should do this check at the beginning of this method, and delegate the entire constructor body to a separate method which is called either immediately or after load.

valdrinkoshi replied:
updated, PTAL

alice commented on inert.js:
```diff
@@ -481,6 +485,21 @@ class InertManager {
+    for (let inertElement of inertElements)
+      this.setInert(inertElement, true);
+
+    // Add inert style.
```
I don't think this needs to wait for document load?

alice commented:
LGTM

I invited you to be a collaborator so you can pull whenever you like - you can decide what you feel about the styles thing :)

valdrinkoshi commented:
@alice I've updated the style to be added in the constructor, it will try to add it in `<head>`, `<body>` or `<html>` node, whichever is available. I was tempted to use `document.currentScript.parentNode`, but [`currentScript`](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) is not supported by IE 😞 
I've updated inert.html too, it imports the script in `<head>` now.

alice commented:
Still LGTM, pull whenever :)
